ambry-frontend/src/main/java/com/github/ambry/frontend/GetBlobHandler.java

```diff
@@ -210,27 +205,76 @@ private Callback<GetBlobResult> routerCallback() {
     private Callback<Void> securityProcessResponseCallback(GetBlobResult result) {
       return buildCallback(metrics.getBlobSecurityProcessResponseMetrics, securityCheckResult -> {
             ReadableStreamChannel response = result.getBlobDataChannel();
-            if (subResource != null && !subResource.equals(SubResource.Segment)) {
-              BlobInfo blobInfo = result.getBlobInfo();
-              if (restRequest.getArgs().containsKey(SEND_USER_METADATA_AS_RESPONSE_BODY) && (boolean) restRequest.getArgs()
-                  .get(SEND_USER_METADATA_AS_RESPONSE_BODY)) {
-                restResponseChannel.setHeader(Headers.CONTENT_TYPE, "application/octet-stream");
-                restResponseChannel.setHeader(Headers.CONTENT_LENGTH, blobInfo.getUserMetadata().length);
-                response = new ByteBufferReadableStreamChannel(ByteBuffer.wrap(blobInfo.getUserMetadata()));
-              } else {
-                restResponseChannel.setHeader(Headers.CONTENT_LENGTH, 0);
-                response = new ByteBufferReadableStreamChannel(EMPTY_BUFFER);
+            if (subResource == null) {
```

Comment:
This refactor is ok since it vastly improves reaadability

Reply:
But I still recommend a separate PR since I just want to focus on your changes alone

Reply:
A bit tedious to separate out what is really your change and what is not.

Reply:
For this part, if I don't optimize the code, with my new changes it will be much harder to read. So I would prefer to use this way instead.
The only new feature I added is case BlobChunkIds.